pagefaultgames/pokerogue
# [QoL] One Pokerus Pokemon for Every Generation

## What are the changes the user will see?
Pokerus is spread out 1 pokemon per gen on pokerus generation.

## Why am I making these changes?
This is a quite highly requested feature and I believe in the reasoning behind it.
-Being able to have a pokerus mon for single gen challenges
-Encouraging people to use more mons spread across gens
-WE NO LONGER HAVE TO CHECK FOR DUPES IN GENERATION LOGIC
-Honestly doesn't break balance especially still randomized
-Again just giga requested :)

## What are the changes from a developer perspective?
-Changed Pokerus cursor objects to 9 instead of 3 (1 per gen)
-Changed generation loop and logic to be gen based and use "this.allspecies" for filtered by gen mons
-NO LONGER DUPE CHECK 😎  (because each of the 9 mons will be from a specific gen so no dupe chance)

### Screenshots/Videos
![image](https://github.com/user-attachments/assets/aa35f4db-e1dd-4612-ab49-574e050ab776)
![image](https://github.com/user-attachments/assets/fb977ac8-c67f-47f2-9941-585f2b814ea9)
![image](https://github.com/user-attachments/assets/c6a7822f-4aae-40ef-a882-c9e1823c20b1)


## How to test the changes?
Basically just boot it up!  
You can see all 9 pokemon whether you've caught them or not, 
but if you want to test a 4 mon party with the rus,
import the everything data and it runs smooth :) 

## Checklist
